tests/core.py

```diff
@@ -1276,6 +1276,11 @@ def test_get_dags(self):
         with self.assertRaises(AirflowException):
             cli.get_dags(self.parser.parse_args(['clear', 'foobar', '-dx', '-c']))
 
+    def test_clear_no_reset(self):
+        args = self.parser.parse_args([
+            'clear', 'example_python_operator', '--no_dag_reset', '--no_confirm'])
+        cli.clear(args)
```

Comment:
I feel that some assertion that the flag does what we expect is needed here please.

(Late review, sorry!)